plugins/modules/docker_container.py

```diff
@@ -2641,6 +2670,26 @@ def _get_expected_binds(self, image):
         self.log(result, pretty_print=True)
         return result
 
+    def _get_expected_labels(self, image):
+        if self.parameters.image_label_mismatch == 'ignore':
```

Comment:
If `labels` hasn't been specified, labels must not be compared at all:
```suggestion
        if self.parameters.labels is None:
            return None
        if self.parameters.image_label_mismatch == 'ignore':
```